cpp/include/cudf/detail/iterator.cuh

```diff
@@ -237,23 +262,26 @@ auto make_null_replacement_pair_iterator(column_device_view const& column,
  * @brief Constructs a pair iterator over a column's values and its validity.
  *
  * Dereferencing the returned iterator for element `i` will return
- * `pair(column[i], true)`. This iterator is only allowed for non-nullable
- * columns.
+ * `pair(column[i], validity)`.
+ * `validity` is `true` if `has_nulls=false`.
+ * `validity` is validity of the element at `i` if `has_nulls=true` and the
+ * column is nullable.
  *
  * @throws `cudf::logic_error` if the column is nullable.
  * @throws `cudf::logic_error` if column datatype and Element type mismatch.
  *
  * @tparam Element The type of elements in the column
+ * @tparam has_nulls boolean indicating to treat the column is nullable
  * @param column The column to iterate
  * @return auto Iterator that returns valid column elements, and validity of the
  * element in a pair
  */
-template <typename Element>
+template <bool has_nulls, typename Element>
```

Comment:
Why not put the template arguments in the same order as in the `pair_accessor` struct?